File: backend/microservice/src/admin/api/director/level_assignment/body/UpdateLevelSlotBirdPoolBody.scala
package microservice.admin.api.director.level_assignment.body

import cats.effect.IO
import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}
import microservice.level.objects.inventory.BirdPool
import org.http4s.EntityDecoder
import org.http4s.circe.jsonOf

/** PUT level-assignments/:levelSuffix/bird-pool 请求体；birdPool 覆盖槽位现有配置。 */
final case class UpdateLevelSlotBirdPoolBody(
  birdPool: BirdPool
)

object UpdateLevelSlotBirdPoolBody {
  implicit val encoder: Encoder[UpdateLevelSlotBirdPoolBody] = deriveEncoder
  implicit val decoder: Decoder[UpdateLevelSlotBirdPoolBody] = deriveDecoder
  implicit val entityDecoder: EntityDecoder[IO, UpdateLevelSlotBirdPoolBody] = jsonOf
}

File: frontend/src/api/admin/director/level_assignment/body/UpdateLevelSlotBirdPoolBody.ts
import { z } from "zod";
import { BirdPoolSchema } from "../../../../../objects/level/inventory/bird-pool.js";

export const UpdateLevelSlotBirdPoolRequestBodySchema = z.object({
  birdPool: BirdPoolSchema,
});

export type UpdateLevelSlotBirdPoolRequestBody = z.infer<typeof UpdateLevelSlotBirdPoolRequestBodySchema>;
